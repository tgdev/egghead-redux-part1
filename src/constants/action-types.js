export const ACTION_ADD_TODO = 'ADD_TODO';
