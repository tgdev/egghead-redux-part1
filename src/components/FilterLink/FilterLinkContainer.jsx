import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { ACTION_SET_VISIBILITY_FILTER } from 'constants/action-types';

import store from '../../store';

import FilterLink from './FilterLink';

class FilterLinkContainer extends Component {
  componentDidMount() {
    store.subscribe(() => {
      this.forceUpdate();
    });
  }

  componentWillUnmount() {
    this.unsubscribe();
  }

  render() {
    const { filter, children } = this.props;
    const { visibilityFilter } = store.getState();
    const isActive = filter === visibilityFilter;

    return (
      <FilterLink
        active={isActive}
        handleFilter={() => {
          store.dispatch({
            type: ACTION_SET_VISIBILITY_FILTER,
            filter
          });
        }}>
        {children}
      </FilterLink>
    );
  }
}

FilterLinkContainer.propTypes = {
  filter: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired
};

export default FilterLinkContainer;
