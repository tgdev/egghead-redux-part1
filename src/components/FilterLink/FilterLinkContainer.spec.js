import React from 'react';
import { shallow, mount } from 'enzyme';

import { FILTER_OPTION_ALL } from 'constants/visibility-filter-options';

import FilterLinkContainer from './FilterLinkContainer';

describe('FilterLinkContainer component', () => {

  const mockFilterOption = FILTER_OPTION_ALL;

  it('renders FilterLinkContainer correctly', () => {
    const wrapper = shallow(
      <FilterLinkContainer
        filter={mockFilterOption}>
        All
      </FilterLinkContainer>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with required children', () => {
    const wrapper = shallow(
      <FilterLinkContainer
        filter={mockFilterOption}>
        All
      </FilterLinkContainer>
    );
    expect(wrapper.children().length).toBe(1);
  });

  it('renders with required filter prop', () => {
    const wrapper = mount(
      <FilterLinkContainer
        filter={mockFilterOption}>
        All
      </FilterLinkContainer>
    );
    const filterProp = wrapper.props().filter;
    expect(filterProp).toEqual(FILTER_OPTION_ALL);
  });

});
