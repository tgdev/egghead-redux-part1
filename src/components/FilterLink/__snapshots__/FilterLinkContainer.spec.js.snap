// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLinkContainer component renders FilterLinkContainer correctly 1`] = `
<FilterLink
  active={true}
  handleFilter={[Function]}
>
  All
</FilterLink>
`;
